website/src/components/Submission/FormOrUploadWrapper.tsx

```diff
@@ -0,0 +1,149 @@
+import { useEffect, useState, type Dispatch, type FC, type SetStateAction } from 'react';
+
+import type { UploadAction } from './DataUploadForm';
+import type { ColumnMapping } from './FileUpload/ColumnMapping';
+import { SequenceEntryUpload } from './FileUpload/SequenceEntryUploadComponent';
+import type { ProcessedFile } from './FileUpload/fileProcessing';
+import type { InputField } from '../../types/config';
+import type { ReferenceGenomesSequenceNames } from '../../types/referencesGenomes';
+import { EditableMetadata, EditableSequences, MetadataForm, SequencesForm } from '../Edit/InputForm';
+
+export type InputMode = 'form' | 'bulk';
+
+export type SequenceData = {
+    type: 'ok';
+    metadataFile: File;
+    sequenceFile?: File;
+};
+
+export type InputError = {
+    type: 'error';
+    errorMessage: string;
+};
+
+export type FileFactory = () => Promise<SequenceData | InputError>;
+
+type FormOrUploadWrapperProps = {
+    inputMode: InputMode;
+    setFileFactory: Dispatch<SetStateAction<FileFactory | undefined>>;
+    organism: string;
+    action: UploadAction;
+    referenceGenomeSequenceNames: ReferenceGenomesSequenceNames;
+    metadataTemplateFields: Map<string, InputField[]>;
+    enableConsensusSequences: boolean;
+};
+
+/**
+ * A component that allows users to upload sequence data. Two modes are supported, see {@link InputMode}.
+ * In 'form' mode, a form is displayed, and the user can directly enter stuff into the form to upload their
+ * metadata. In 'bulk' mode, the user needs to upload files containing the sequences and metadata.
+ * Either way, the component turns the uploaded data into files, so they can be submitted to the API.
+ * Set the 'fileCreatorSetter' to get the files - have a look at existing usage on how this works.
```

Comment:
Should this be `setFileFactory`

Reply:
yes, fixed: https://github.com/loculus-project/loculus/pull/3710/commits/1788f4aa7b1863a8413d0c1b5c8bdd7b6f6ba475